### PR TITLE
String to string conversion

### PR DESCRIPTION
**Pull request by atacan**
* Test was also added.
* I also noticed `strong` was missing from the decoder. It is added, too.

### CODE REVIEW

**Comment by mattesmohr** [comment]
Thanks for your effort. I will check it tomorrow.

File: Sources/HTMLKit/Internal/Features/Conversion/Converter.swift [hunk 1]
```diff
@@ -104,6 +104,22 @@ public class Converter {
         }
     }
     
+    /// Converts an HTML component to be put into an existing Page or View layout.
+    /// The whole html string needs to be inside a tag.
+    /// For example, multiple `div`s will give an error. They need to be inside another `div`.
+    public func convert(html: String) throws -> String {
+        // The converter puts a new line inside a single quote Swift string, which breaks the code. To avoid that:
```
Comment (mattesmohr) [inline Sources/HTMLKit/Internal/Features/Conversion/Converter.swift:111]: If the code is not valid, the converter sees the content as a single text node. Do you mean it by that?

**Comment by atacan** [comment]
I had to remove the `\n` because the following html 
```html
<h1>
This is heading 1
</h1>
```
gives
```swift
Heading1 {
	"
This is heading 1
"
}
```

**Comment by atacan** [comment]
The first explanation is about enclosing the html code inside one tag.
Because 
```html
<h1>This is heading 1</h1>
<h1>This is heading 1</h1>
```
gives `Line 2: Extra content at the end of the document` error. So, the whole code should be wrapped by a tag.

**Comment by atacan** [comment]
This problem that I described above as "a new line inside a single quote Swift string" happens on the file conversion, too.
Removing `\n`s solves the problem for now, but we lose the line number information `XMLDocument` returns when there is an error.

**Comment by atacan** [comment]
I fixed it in a better way. I handle the new line in the TextComponent
https://github.com/vapor-community/HTMLKit/pull/78/commits/00cf08c32c4eac4d7bd9d57435eb599ab0ef4f99

**Comment by mattesmohr** [comment]
@atacan I am currently traveling. I will review tomorrow. Did you see my comment about the test function?

**Comment by atacan** [comment]
Hello, you can go ahead and modify the tests. Thank you

File: Sources/HTMLKit/Internal/Features/Conversion/Converter.swift [hunk 2]
```diff
@@ -645,7 +644,12 @@ extension Converter {
             let indent = String(repeating: "\t", count: (node.level - 1) + (preindent ?? 0))
             
             if let text = text {
-                "\(indent)\"\(text)\"\n"
+                if node.parent?.localName != "pre" {
```
Comment (mattesmohr) [inline Sources/HTMLKit/Internal/Features/Conversion/Converter.swift]: What's this for? And is it a specific case of the string to string conversion or would it be necessary for the decoding in general?

File: Sources/HTMLKit/Internal/Features/Conversion/Converter.swift [hunk 3]
```diff
@@ -645,7 +644,12 @@ extension Converter {
...
+                    let cleanText = text.replacingOccurrences(of: "\n", with: "")
+                    "\(indent)\"\(cleanText)\"\n"
+                } else {
+                    "\(indent)\"\"\"\(text)\"\"\"\n"
+                }
```
Comment (mattesmohr) [inline Sources/HTMLKit/Internal/Features/Conversion/Converter.swift:652]: What's this for? And is it a specific case of the string to string conversion or would it be necessary for the decoding in general?

File: Sources/HTMLKit/Internal/Features/Conversion/Converter.swift [hunk 4]
```diff
@@ -644,11 +644,11 @@ extension Converter {
...
             if let text = text {
-                if node.parent?.localName != "pre" {
-                    let cleanText = text.replacingOccurrences(of: "\n", with: "")
-                    "\(indent)\"\(cleanText)\"\n"
-                } else {
+                if node.parent?.localName = "pre" {
                     "\(indent)\"\"\"\n\(text)\"\"\"\n"
+                } else {
+                    let cleanText = text.trimmingCharacters(in: .whitespacesAndNewlines)
+                    "\(indent)\"\(cleanText)\"\n"
```
Comment (mattesmohr) [inline Sources/HTMLKit/Internal/Features/Conversion/Converter.swift:651]: What's the """ in a pre for?

Comment (mattesmohr) [reply Sources/HTMLKit/Internal/Features/Conversion/Converter.swift:651]: Can you give me a sample of code you try to convert, so I can see what's happening here?

Comment (atacan) [reply Sources/HTMLKit/Internal/Features/Conversion/Converter.swift:651]:
```html
<body>
  <h1>Header</h1>
  <h1>      Hello      World!     </h1>
  <p>
    Paragraph
  </p>
<pre>struct ContentView: View {
    var body: some View {
        AdaptiveStack {
            Text("Horizontal when there's lots of space")
            Text("but")
            Text("Vertical when space is restricted")
        }
    }
}
</pre>
</body>
```

Comment (atacan) [reply Sources/HTMLKit/Internal/Features/Conversion/Converter.swift:651]: we preserve the whitespace and the new lines in `pre` and remove them in the other tags.
I was reading [this].(https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Whitespace)
So, unless it is in `pre` we should remove all the new lines.
But I just noticed the built-in function trims it only from the beginning and end. I need to do one more commit for non-`pre` case.

Comment (atacan) [reply Sources/HTMLKit/Internal/Features/Conversion/Converter.swift:651]: This Html snippet covers the cases well
```html
<!DOCTYPE html>
<html>
<body>
  <h1>Header</h1>
  <h1>      Hello      
  World!     </h1>
  <p>
    Paragraph
  </p>
<pre>private var comment: String? {
            
            guard let comment = node.stringValue else {
                return nil
            }
            
            return comment
        }
</pre>
<p>struct ContentView: View {
    var body: some View {
        AdaptiveStack {
            Text("Horizontal when there's lots of space")
            Text("but")
            Text("Vertical when space is restricted")
        }
    }
}
</p>
</body>
</html>
```

Comment (atacan) [reply Sources/HTMLKit/Internal/Features/Conversion/Converter.swift:651]: In my view this should be the output. What do you think?
```swift
Html {
            Body {
                Heading1 {
                    "Header"
                }
                Heading1 {
                    "Hello        World!"
                }
                Paragraph {
                    "Paragraph"
                }
                PreformattedText {
                    """
        private var comment: String? {
                    
                    guard let comment = node.stringValue else {
                        return nil
                    }
                    
                    return comment
                }
        """
                }
                Paragraph {
                    #"struct ContentView: View {    var body: some View {        AdaptiveStack {            Text("Horizontal when there's lots of space")            Text("but")            Text("Vertical when space is restricted")        }    }}"#
                }
            }
        }
```

**Review by mattesmohr — APPROVED** [review]
Looks good to me.

**Pull request merged**